src/Database/Type/DateTimeType.php

```diff
@@ -53,6 +53,8 @@ class DateTimeType extends BaseType implements BatchCastingInterface
         'Y-m-d\TH:i',
         'Y-m-d\TH:i:s',
         'Y-m-d\TH:i:sP',
+        'Y-m-d H:i:s.P',
+        'Y-m-d H:i:s.u',
```

Comment:
A datetime column with microseconds should be mapped to the `DateTimeFractionalType` class, which already has this [format](https://github.com/cakephp/cakephp/blob/5.x/src/Database/Type/DateTimeFractionalType.php#L41) so it's not needed here.

Reply:
That was the original assumption. However, DAteTimeFractionalType will create a string with the fraction and that probably breaks the mysql parser.

Reply:
If you think this is fine/necessary then I have no issues. I haven't used microseconds in practice so not familiar with it's intricacies.

Reply:
It's definitely unexpected input, but what can you do.

In the future, I think we might simplify these types so they parse anything reasonable and only care about how they convert to db format and timezone.

Reply:
As explained in the ticket, this about the datetime value part of the value, not the milliseconds.
The expectation is that this first part is not dropped silently.
That the ms cannot be stored is fine here.

Reply:
I think it makes sense to parse the microseconds into the PHP object but not persist that precision.